frontend/src/metabase/query_builder/components/view/ViewHeader/components/DashboardBackButton/DashboardBackButton.tsx

```diff
@@ -29,8 +39,12 @@ export function DashboardBackButton() {
     <Tooltip tooltip={label}>
       <BackButtonContainer>
         <BackButton
-          as={Link}
-          to={Urls.dashboard(dashboard)}
+          {...(noLink
```

Comment:
Is there a scenario where we would want to apply onClick without also applying noLink? It seems to me like we would always require `noLink` to be `true` anytime an onClick handler is provided, so the condition here could also be `onClick ? {} : ...`.

I think the current code is fine if we want to be explicit about disabling links though.